custom_components/test/fuzzing/fuzzer_utils/GeneratorFuzzer.py

```diff
@@ -0,0 +1,216 @@
+        :type data_with_marker: Union[Tuple[str, frozenset], Any]
+        :return: The original data form if 'data_with_marker' had a type marker, otherwise 'data_with_marker' itself.
+        :rtype: Any
+        """
+        # Check if data_with_marker is a tuple and has a type marker
+        if isinstance(data_with_marker, tuple) and data_with_marker[0] in ('list', 'dict'):
+            data_type, data = data_with_marker
+            if data_type == 'list':
+                return list(data)
+            elif data_type == 'dict':
+                return dict(data)
+        else:
+            # If there is no type marker, return the data as is
+            return data_with_marker
+
+    def fuzz(self, cls: Type, start_methods: List[str], max_sequence_length: int, num_sequences: int, max_param_combi: int = 2) -> List[List[Tuple[str, Tuple[Any]]]]:
+        """
+        Generate unique method sequences with parameters. A sequence is unique if either the methods are chained differently or any value of any parameter differs from an otherwise identical sequence.
+
+        :param cls: The class for which the method sequences are being generated.
+        :type cls: Type
+        :param start_methods: List of method names that can start the sequence.
+        :type start_methods: List[str]
+        :param max_sequence_length: The maximum length of the method sequences.
+        :type max_sequence_length: int
+        :param num_sequences: The number of unique sequences to generate.
+        :type num_sequences: int
+        :param max_param_combi: Maximum number of parameter combinations. If a method takes less parameters the combinations will be decreased accordingly.
+        :type param_combi: int
+        :return: List of unique method sequences with parameters.
+        :rtype: List[List[Tuple[str, Tuple[Any]]]]
+        example return:
+        [
+            [('method1', (param1, param2)), ('method2', (param3, param4))],
+            [('method3', (param5)), ('method4', (param6, param7))]
+        ]  
+        """
+        existing_sequences = set()
+        while len(existing_sequences) < num_sequences:
+            sequence = self._generate_method_sequence(cls, start_methods, max_sequence_length)
+            param_sequences = []
+            for method_name in sequence:
+                param_combi = max_param_combi
+                method = getattr(cls, method_name)
+                param_types = self._get_param_types(cls.__name__, method)
+                if len(param_types) != 0:
+                    # make sure we don't request more combinations than parameters available
+                    if max_param_combi > len(param_types):
+                        param_combi = len(param_types)
+                    # Generate parameter sets using the value pool fuzzer 
+                    param_set = self.value_pool_fuzzer.fuzz(param_nr=len(param_types), types=[param_types[i] for i in range(len(param_types))], param_combi=param_combi)
```

Comment:
`param_nr` is not longer a parameter of the `value_pool_fuzzer` `fuzz()` function.
See run-time error:

` TypeError: ValuePoolFuzzer.fuzz() got an unexpected keyword argument 'param_nr'`